File: 3_Arrays/Sortings/insertion sort/insertion sort.c
#include <stdio.h>


int main()
{

    const int ARRAYLEN = 8;
    int array[] = {10, 3, 6, 2, 6, 8, 1, 45, 7};
    int num, j;

    for (int i = 1; i < ARRAYLEN; i++)
    {
        num = array[i];
        j = i - 1;

        while (j >= 0 && array[j] > num)
        {
            array[j + 1] = array[j];
            j--;
        }
        array[j + 1] = num;
    }

    for (int i = 0; i < ARRAYLEN; i++)
        printf("%d ", array[i]);

    return 0;
}

File: 3_Arrays/Sortings/insertion sort/insertion_sort.py
def insertion_sort(array):
    for i in range(1, len(array)):
        num = array[i]
        j = i - 1
        while j >= 0 and num < array[j]:
            array[j + 1] = array[j]
            j -= 1
        array[j + 1] = num
    return array

# Driver code
to_sort = [0, 1, 4, 6, 2, 6, 1, 9, 12]
print(insertion_sort(to_sort))
